src/js/jsx/shared/Datalist.jsx

```diff
@@ -281,65 +328,226 @@ define(function (require, exports, module) {
+                    if (idString.indexOf(word.toLowerCase()) > -1) {
+                        return "";
+                    }
+                    return word;
+                });
+
+            var nextFilter = nextFilterMap.join(" ").trim();
+
+            this._updateAutofill(nextFilter, iconCount);
+
+            if (this.props.startFocused) {
+                this.refs.textInput._beginEdit(false);
+            }
+        },
+
         render: function () {
             // HACK - Because Select has no correspondence between ID and title
             // during selection methods, only when the Datalist component is not live
             // we manually set the shown value of the input to the selected option's title
             // This can be an expensive operation when options is big enough, so 
-            // use carefully.
-            var current;
+            // use carefully. If we are using autocomplete, then we can use the suggestion
+            // title, since it corresponds with the current ID.
+            var current,
+                currentTitle = this.props.value;
+            
             if (!this.props.live) {
-                current = this.props.options.find(function (option) {
-                    return option.id === this.state.id;
-                }.bind(this));
+                if (this.props.useAutofill && this.state.suggestTitle !== "") {
+                    currentTitle = this.state.suggestTitle;
+                } else {
+                    current = this.props.options.find(function (option) {
+                        return option.id === this.state.id;
+                    }.bind(this));
+
+                    currentTitle = current ? current.title : this.props.value;
+                }
             }
 
-            var live = this.props.live,
-                currentTitle = current ? current.title : this.props.value,
-                value = (live ? this.props.value : currentTitle) || "",
+            var value = currentTitle || "",
                 filter = this.state.filter,
                 title = this.state.active && filter !== null ? filter : value,
                 searchableFilter = filter ? filter.toLowerCase() : "",
-                options = this.props.options,
-                searchableOptions = options && options.filter(function (option) {
-                    // Always add headers to list of searchable options
-                    if (option.type && option.type === "header") {
-                        return true;
+                searchableOptions = this._filterOptions(searchableFilter);
+            
+            var autocomplete,
+                hiddenTI;
+
+            if (this.props.useAutofill) {
+                hiddenTI = (
+                    <div ref="hiddenTextInput"
+                        className="hiddeninput">
+                    </div>
+                );
+                
+                var autocompStyle = { left: this.state.width + "px" },
```

Comment:
Yeah, so this is adjusting the positioning of the suggestion text. As the user types, the text needs to move over letter by letter.

Maybe calling it `position` makes mores sense?